admin/create-theme/theme-readme.php

```diff
@@ -1,50 +1,103 @@
-		$is_parent_theme     = $theme['is_parent_theme'] ?? false;
-		$original_theme      = $theme['original_theme'] ?? '';
 
-		// Handle copyright section.
-		$new_copyright_section  = $is_parent_theme || $original_theme ? true : false;
-		$original_theme_credits = $new_copyright_section ? self::original_theme_credits( $name, $is_parent_theme ) : '';
-		$copyright_section      = self::copyright_section( $new_copyright_section, $original_theme_credits, $name, $copy_year, $author, $image_credits );
+	/**
+	 * Get the path to the readme.txt file.
+	 *
+	 * @return string
+	 */
+	public static function file_path() {
+		return path_join( get_stylesheet_directory(), 'readme.txt' );
+	}
 
-		// Handle recommended plugins section.
-		$recommended_plugins_section = self::recommended_plugins_section( $recommended_plugins ) ?? '';
+	/**
+	 * Get the content of the readme.txt file.
+	 *
+	 * @return string
+	 */
+	public static function get_content() {
+		$path = self::file_path();
+		if ( ! file_exists( $path ) ) {
+			return '';
+		}
+		return file_get_contents( $path );
+	}
 
-		return "=== {$name} ===
+	/**
+	* Build a readme.txt file for CHILD/GRANDCHILD themes.
+	*
+	* @param array $theme The theme data.
+	* {
+	*     @type string $name The theme name.
+	*     @type string $description The theme description.
+	*     @type string $uri The theme URI.
+	*     @type string $author The theme author.
+	*     @type string $author_uri The theme author URI.
+	*     @type string $copyright_year The copyright year.
+	*     @type string $image_credits The image credits.
+	*     @type string $recommended_plugins The recommended plugins.
+	*     @type bool $is_child_theme Whether the theme is a child theme.
+	* }
+	*
+	* @return string The readme content.
+	*/
+	public static function create( $theme ) {
+		$name                 = $theme['name'];
+		$description          = $theme['description'] ?? '';
+		$uri                  = $theme['uri'] ?? '';
+		$author               = $theme['author'] ?? '';
+		$author_uri           = $theme['author_uri'] ?? '';
+		$copy_year            = $theme['copyright_year'] ?? gmdate( 'Y' );
+		$wp_version           = $theme['wp_version'] ?? get_bloginfo( 'version' );
+		$required_php_version = $theme['required_php_version'] ?? '5.7';
+		$license              = $theme['license'] ?? 'GPLv2 or later';
+		$license_uri          = $theme['license_uri'] ?? 'http://www.gnu.org/licenses/gpl-2.0.html';
+		$image_credits        = $theme['image_credits'] ?? '';
+		$recommended_plugins  = $theme['recommended_plugins'] ?? '';
+		$is_child_theme       = $theme['is_child_theme'] ?? false;
+
+		// Generates a reference to the original theme for child and cloned themes.
+		$original_theme_credits = $is_child_theme
+			? self::original_theme_credits( $name, $is_child_theme )
+			: null;
+		// Generates the copyright section text.
+		$copyright_section_content = self::copyright_text( $name, $copy_year, $author, $original_theme_credits );
+
```

Comment:
I find this part very confusing.  How could we refactor this to be more clear what is happening and why?

Reply:
Perhaps we could make this 
```
$copyright_section_content = self::generate_copyright_text( $theme );
```

And within that function determine what copyright is most appropriate.

Reply:
Also the comment notes that it would generate a reference for child AND cloned themes, however the "cloned" part of this seems to have gone away.

Reply:
Thanks for flagging that. I simplified it in this commit: https://github.com/WordPress/create-block-theme/pull/626/commits/904f45e06bd46f7c424a0bc7bb1370cb24781d1a

Reply:
I fixed the cloned themes reference and added a test case for it here: https://github.com/WordPress/create-block-theme/pull/626/commits/b4cb121893f986f185de99044de2bf3308f55a03

Reply:
This is much easier to understand!